django/contrib/admin/checks.py

```diff
@@ -333,9 +333,13 @@ def _check_fields(self, obj):
             ]
         fields = flatten(obj.fields)
         if len(fields) != len(set(fields)):
+            duplicated_fields = list(
+                sorted({field for field in fields if fields.count(field) > 1})
+            )
             return [
                 checks.Error(
                     "The value of 'fields' contains duplicate field(s).",
+                    hint="Remove duplicates of these field(s): %s." % duplicated_fields,
```

Comment:
Favor the usage of a `collections.Counter` for these kind of things to avoid scanning `fields` multiple times

```suggestion
        fields = Counter(flatten(obj.fields))
        if duplicate_fields := sorted(field for field, count in fields.items() if count > 1):
            return [
                checks.Error(
                    "The value of 'fields' contains duplicate field(s).",
                    hint="Remove duplicates of these field(s): %s." % duplicated_fields,
```